lib/components/NumberedCircle.dart

```diff
@@ -18,26 +18,24 @@ class NumberedCircle extends StatelessWidget {
 }
 
 class Circle extends StatelessWidget {
-  const Circle({required this.size, required this.color, this.text, super.key});
+  const Circle({required this.size, required this.color, this.text = '', super.key});
 
   final double size;
   final Color color;
-  final String? text;
+  final String text;
 
   @override
   Widget build(BuildContext context) {
     return Container(
       width: size,
       height: size,
       decoration: $styles.containerStyles.circular.copyWith(color: color),
-      child: text != null
-          ? Center(
-              child: Text(text!,
-                  style: text!.length > 2
-                      ? $styles.text.heading3
-                      : $styles.text.heading2),
-            )
-          : null,
+      child: Center(
+        child: Text(
+          text,
+          style: text.length > 2 ? $styles.text.heading3 : $styles.text.heading2,
```

Comment:
Just simple reductions here based on text not being nullable and having a fallback value.

Reply:
I've been writing too many null checks that I didn't need to write -.-